### PR TITLE
[4.0] Trigger the onBeforeExecute correctly through the dispatcher

### PR DESCRIPTION
**Pull request by laoneo**
Pull Request for Issue #20435.

### Summary of Changes
Sends the `onBeforeExecute` correctly in the applications.

### Testing Instructions
Enable the language filter plugin.

### Expected result
No exception is thrown.

### Actual result
Exception 
_Argument 1 passed to PlgSystemLanguageFilter::onBeforeExecute() must be an instance of Joomla\CMS\Event\BeforeExecuteEvent, instance of Joomla\Event\Event given_
is thrown.

### CODE REVIEW

**Comment by infograf768** [comment]
I have tested this item :white_check_mark: successfully on 3fbe0426ebbb8e32f4342f2ecf42bdc44ec9a443<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20448">issues.joomla.org/tracker/joomla-cms/20448</a>.</sub>

**Comment by infograf768** [comment]
@mbabker @wilsonge This one is rather urgent if we want to test associations and solve the multilingual issues

**Comment by roland-d** [comment]
I have tested this item :white_check_mark: successfully on ac6b1796df758d525ba8bec91ebc4dc966d4f015<br /><br />Before the fix after I activated the plugin it was impossible to access Joomla, after the fix everything worked as expected.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/20448">issues.joomla.org/tracker/joomla-cms/20448</a>.</sub>

**Comment by ghost** [comment]
@infograf768 can you please retest?

File: libraries/src/Application/WebApplication.php [hunk 1]
```diff
@@ -128,8 +129,11 @@ public static function getInstance($name = null)
 	 */
 	public function execute()
 	{
-		// Trigger the onBeforeExecute event.
-		$this->triggerEvent('onBeforeExecute');
+		// Trigger the onBeforeExecute event
+		$this->getDispatcher()->dispatch(
+			'onBeforeExecute',
+			new BeforeExecuteEvent('onBeforeExecute', ['subject' => $this, 'container' => $this->getContainer()])
```
Comment (wilsonge) [inline libraries/src/Application/WebApplication.php:135]: This one doesn't work sadly :( The `getContainer` doesn't exist in this class

Comment (rdeutz) [reply libraries/src/Application/WebApplication.php:135]: I made a PR to fix it

**Pull request merged**